nemo/collections/asr/parts/utils/streaming_utils.py

```diff
@@ -1672,12 +1700,102 @@ def transcribe(
         """
         self.infer_logits(keep_logits)
 
-        hypothesis = " ".join([h.text for h in self.all_preds])
+        # join hypotheses 
+        hypothesis = self._join_hypotheses(self.all_preds)
```

Comment:
instead of this `predictions = self.asr_model.predict_step(batch_input, has_processed_signal=True)` why not use `process_aed_timestamp_outputs` with an option to pass offset to it? basically combination of `_transcribe_forward` + `_transcribe_output_processing`